src/stores/nodeParamStore.ts

```diff
@@ -0,0 +1,273 @@
+// should apply this to litegraph node
+import { defineStore } from 'pinia'
+import { ref, computed, watch } from 'vue'
+import { app } from '@/scripts/app'
+
+// todo: Use Litegraph.js types here when we have a Litegraph ts version or Zod Schema
+interface Widget {
+  title: string
+  type: string
+  label: string
+  options: any
+  value: object | string | number | boolean | Array<any> | null
+  fav: boolean
+}
+
+interface Node {
+  title: string
+  id: number
+  order: number
+  color: string
+  bypass: boolean
+  bgcolor: string
+  mode: number
+  favWidgets: FavWidgets[]
+  category: string
+  widgets: Widget[]
+}
+
+interface FavWidgets {
+  name: string
+  fav: boolean
+}
+
+// interface Gallery {
+//     images: string[];
+// }
+
+interface category {
+  name: string
+  count: number
+}
+
+export const useNodeParamStore = defineStore('nodeParam', () => {
+  // State
+  const nodes = ref<Node[]>([])
+  const widgetFav = ref<Widget[]>([])
+  let filterNodes = ref(nodes)
+  const categories = computed<category[]>(() => {
+    let favCount = 0
+    for (const node of nodes.value) {
+      for (const widget of node.widgets)
+        if (widget.fav) {
+          favCount++
+        }
+    }
+    return [
+      { name: 'All', count: nodes.value.length },
+      { name: 'Star', count: favCount },
+      {
```

Comment:
Is this notnull category really needed? What would be the use case?

Reply:
Since many 'Reroute' node would cause the difficulty of finding target node